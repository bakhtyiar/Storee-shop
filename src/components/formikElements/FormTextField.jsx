import React from "react";
import {Form, InputGroup} from "react-bootstrap";
import {Field} from "formik";

const FormTextField = ({
                           as,
                           md,
                           controlId,
                           label,
                           name,
                           type,
                           inputGroupPrepend,
                           placeholder,
                           formBottomText
                       }) => {
    return (
        <Field name={name}>
            {({field, form}) => {
                const isValid = !form.errors[field.name];
                const isInvalid = form.touched[field.name] && !isValid;
                return (
                    <Form.Group as={as} md={md} controlId={controlId} data-testid={"form-text-field"}>
                        <Form.Label>{label}</Form.Label>
                        <InputGroup>
                            {inputGroupPrepend}
                            <Form.Control
                                className={'rounded'}
                                {...field}
                                type={type}
                                isValid={form.touched[field.name] && isValid}
                                isInvalid={isInvalid}
                                feedback={form.errors[field.name]}
                                placeholder={placeholder}
                            />

                            <Form.Control.Feedback type="invalid" data-testid="error-feedback">
                                {form.errors[field.name]}
                            </Form.Control.Feedback>
                        </InputGroup>
                        {formBottomText && (
                            <>
                                <Form.Text className="text-muted">
                                    {formBottomText}
                                </Form.Text>
                            </>
                        )}
                    </Form.Group>
                );
            }}
        </Field>
    );
};

FormTextField.defaultProps = {
    type: "text",
    inputGroupPrepend: null
};

export default FormTextField;
