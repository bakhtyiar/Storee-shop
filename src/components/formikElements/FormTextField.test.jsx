import FormTextField from "./FormTextField";
import {screen} from "@testing-library/react";
import {renderWithFormikContext} from "../../../tests/integration-helpers/renderWithFormikContext";

describe("Basic tests", () => {
    test("Match snapshot", () => {
        renderWithFormikContext(<FormTextField/>);
        expect(screen.getByTestId("form-text-field")).toMatchSnapshot();
    })
})