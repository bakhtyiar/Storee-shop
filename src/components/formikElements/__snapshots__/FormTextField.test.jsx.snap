// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic tests Match snapshot 1`] = `
<div
  data-testid="form-text-field"
>
  <label
    class="form-label"
  />
  <div
    class="input-group"
  >
    <input
      class="rounded form-control"
      type="text"
      value=""
    />
    <div
      class="invalid-feedback"
    />
  </div>
</div>
`;
